expr/func_aggregate.go

```diff
@@ -37,19 +52,38 @@ func (s *FuncAggregate) Exec(dataMap DataMap) ([]models.Series, error) {
 		return nil, err
 	}
 
+	agg := seriesAggregator{function: getCrossSeriesAggFunc(s.name), name: s.name}
+	series = Normalize(dataMap, series)
+	output := aggregate(series, queryPatts, agg, s.xFilesFactor)
+
+	dataMap.Add(Req{}, output)
+
+	return []models.Series{output}, nil
+}
+
+func aggregate(series []models.Series, queryPatts []string, agg seriesAggregator, xFilesFactor float64) models.Series {
 	if len(series) == 0 {
-		return series, nil
+		return models.Series{}
```

Comment:
This is not the right thing to return on empty input. We should not call this function with 0 series.